pkg/morph/deploy/deploy.go

```diff
@@ -256,9 +331,321 @@ func Deploy(ctx context.Context, prm Prm) error {
 
 	prm.Logger.Info("on-chain NNS contract successfully updated")
 
-	// TODO: deploy/update other contracts
+	syncPrm := syncNeoFSContractPrm{
+		logger:            prm.Logger,
+		blockchain:        prm.Blockchain,
+		monitor:           monitor,
+		localAcc:          prm.LocalAccount,
+		nnsContract:       nnsOnChainAddress,
+		systemEmail:       prm.NNS.SystemEmail,
+		committee:         committee,
+		committeeGroupKey: committeeGroupKey,
+	}
+
+	localAccLeads := localAccCommitteeIndex == 0
+
+	var notaryDisabledExtraUpdateArg bool
+
+	// then go dependent contracts. Things may become better/easier after
+	// https://github.com/nspcc-dev/neofs-contract/issues/325
+	resolveContractAddressDynamically := func(commonPrm CommonDeployPrm, contractDomain string) (util.Uint160, error) {
+		domain := calculateContractAddressDomain(contractDomain)
+		onChainState, err := readContractOnChainStateByDomainName(prm.Blockchain, nnsOnChainAddress, domain)
+		if err != nil {
+			// contract may be deployed but not registered in the NNS yet
+			if localAccLeads && (errors.Is(err, errMissingDomain) || errors.Is(err, errMissingDomainRecord)) {
+				return state.CreateContractHash(prm.LocalAccount.ScriptHash(), commonPrm.NEF.Checksum, commonPrm.Manifest.Name), nil
+			}
+			return util.Uint160{}, fmt.Errorf("failed to read on-chain state of the contract by NNS domain '%s': %w", domain, err)
+		}
+		return onChainState.Hash, nil
+	}
+
+	// Alphabet
+	syncPrm.localNEF = prm.AlphabetContract.Common.NEF
+	syncPrm.localManifest = prm.AlphabetContract.Common.Manifest
+	syncPrm.anyValidExtraDeployArgs = []interface{}{false, util.Uint160{}, util.Uint160{}, "az", 0, 0}
+	syncPrm.buildVersionedExtraUpdateArgs = func(versionOnChain contractVersion) ([]interface{}, error) {
+		if versionOnChain.equals(0, 17, 0) {
```

Comment:
iiuc ur question: because on-update `_deploy` callback is executed with on-chain version which may require some args (like this one). If `_deploy` throws exception then contract isn't updated, right?

Reply:
> on-update _deploy callback is executed with on-chain version

Why? It's called _after_ the update. If you mean arguments, then yes, that's what our update methods do (add on-chain version to control upgrades), but the `_deploy` implementation is going to be from the new contract and you need to provide arguments valid for this new version, not the one you have before the upgrade.

Reply:
that's a suprise to me, will fix

Reply:
done